# Primeira Análise - Bank Customer Churn Prediction

# Fonte:
https://www.kaggle.com/kmalit/bank-customer-churn-prediction
 

In [2]:
import pandas as pd

df = pd.read_csv('Churn_Modelling.csv', delimiter=',')
df.shape

(10000, 14)

In [3]:
df.head(5)

RowNumber  CustomerId   Surname  CreditScore Geography  Gender  Age  \
0          1    15634602  Hargrave          619    France  Female   42   
1          2    15647311      Hill          608     Spain  Female   41   
2          3    15619304      Onio          502    France  Female   42   
3          4    15701354      Boni          699    France  Female   39   
4          5    15737888  Mitchell          850     Spain  Female   43   

   Tenure    Balance  NumOfProducts  HasCrCard  IsActiveMember  \
0       2       0.00              1          1               1   
1       1   83807.86              1          0               1   
2       8  159660.80              3          1               0   
3       1       0.00              2          0               0   
4       2  125510.82              1          1               1   

   EstimatedSalary  Exited  
0        101348.88       1  
1        112542.58       0  
2        113931.57       1  
3         93826.63       0  
4         79084.10       0

In [4]:
df.info

<bound method DataFrame.info of       RowNumber  CustomerId    Surname  CreditScore Geography  Gender  Age  \
0             1    15634602   Hargrave          619    France  Female   42   
1             2    15647311       Hill          608     Spain  Female   41   
2             3    15619304       Onio          502    France  Female   42   
3             4    15701354       Boni          699    France  Female   39   
4             5    15737888   Mitchell          850     Spain  Female   43   
...         ...         ...        ...          ...       ...     ...  ...   
9995       9996    15606229   Obijiaku          771    France    Male   39   
9996       9997    15569892  Johnstone          516    France    Male   35   
9997       9998    15584532        Liu          709    France  Female   36   
9998       9999    15682355  Sabbatini          772   Germany    Male   42   
9999      10000    15628319     Walker          792    France  Female   28   

      Tenure    Balance  NumOfP

# Visão geral dos atributos

    - Colunas desnecessárias:
       - RowNumber, CustomerId e Surname. Totalizando no final 11 atributos.
    CreditScore: 350 e 850.
    Geography: 3 países: Alemanha, Espanha e França.
    Gênero: 55% Homens e 45% Mulheres.
    Idade: 18 a 92. Com média de 39 anos.
    Tenure: Tempo que o cliente está no banco: 0 até 10 anos.
    Balance: Saldo bancário do cliente: 0 até 99.986
    NumOfProducts: número de produto do banco que o cliente utiliza. 1 até 4.
    HasCrCard: possuí cartão? Sim ou Não.
    IsActiveMember:Conta ativa ou não.
    EstimatedSalary: Salário estimado: 523 até 99984
    Exited: 1 ou 0 (*Confirmar)
    

In [3]:
# Verificação da existência de valores nulos
df.isnull().sum()

RowNumber          0
CustomerId         0
Surname            0
CreditScore        0
Geography          0
Gender             0
Age                0
Tenure             0
Balance            0
NumOfProducts      0
HasCrCard          0
IsActiveMember     0
EstimatedSalary    0
Exited             0
dtype: int64

In [4]:
# Clientes de diversos países
df['Geography'].unique()


array(['France', 'Spain', 'Germany'], dtype=object)

In [5]:
# Remoção das colunas desnecessárias
df = df.drop(["RowNumber", "CustomerId", "Surname"], axis = 1)

In [6]:
df.describe()

CreditScore           Age        Tenure        Balance  NumOfProducts  \
count  10000.000000  10000.000000  10000.000000   10000.000000   10000.000000   
mean     650.528800     38.921800      5.012800   76485.889288       1.530200   
std       96.653299     10.487806      2.892174   62397.405202       0.581654   
min      350.000000     18.000000      0.000000       0.000000       1.000000   
25%      584.000000     32.000000      3.000000       0.000000       1.000000   
50%      652.000000     37.000000      5.000000   97198.540000       1.000000   
75%      718.000000     44.000000      7.000000  127644.240000       2.000000   
max      850.000000     92.000000     10.000000  250898.090000       4.000000   

         HasCrCard  IsActiveMember  EstimatedSalary        Exited  
count  10000.00000    10000.000000     10000.000000  10000.000000  
mean       0.70550        0.515100    100090.239881      0.203700  
std        0.45584        0.499797     57510.492818      0.402769  
min        0.00000        0.000000        11.580000      0.000000  
25%        0.00000        0.000000     51002.110000      0.000000  
50%        1.00000        1.000000    100193.915000      0.000000  
75%        1.00000        1.000000    149388.247500      0.000000  
max        1.00000        1.000000    199992.480000      1.000000

In [7]:
df.groupby(['Exited']).count()

CreditScore  Geography  Gender   Age  Tenure  Balance  NumOfProducts  \
Exited                                                                         
0              7963       7963    7963  7963    7963     7963           7963   
1              2037       2037    2037  2037    2037     2037           2037   

        HasCrCard  IsActiveMember  EstimatedSalary  
Exited                                              
0            7963            7963             7963  
1            2037            2037             2037

In [8]:
df.dtypes

CreditScore          int64
Geography           object
Gender              object
Age                  int64
Tenure               int64
Balance            float64
NumOfProducts        int64
HasCrCard            int64
IsActiveMember       int64
EstimatedSalary    float64
Exited               int64
dtype: object

# Balanciamento das amostras<br>
Utilização do SMOTE (técnica de sobreamostragem minoritária sintética )

Publicação do Artigo através www.overleaf.com
<br>
*Verificar o layout da FACENS.



Datas importantes:
<br>
Primeira versão do Artigo: 13/11
<br>
Feedback: 27/11
<br>
Versão final e repositório Github: 18/12

